lib/job.js

```diff
@@ -368,14 +368,14 @@ Job.prototype.finished = function(){
         //
         interval = setInterval(function(){
           scripts.isFinished(_this).then(function(status){
-            var finished = status > 0;
-            if(finished){
+            Job.fromId(_this.queue, _this.id).then(function(job){
+              removeListeners();
```

Comment:
I am thinking that maybe it is better to call removeListeners() on a .finally() call on ```scripts.isFinished```, and return Job.fromId so that the errors propagate correctly. A catch with a ```console.error()``` would also be nice :).